WordPress/Classes/ViewRelated/Blog/Site Picker/SiteSwitcherView.swift

```diff
@@ -28,6 +29,20 @@ final class SiteSwitcherViewController: UIHostingController<SiteSwitcherView>, U
         if #available(iOS 17, *) {
             AppTips.SitePickerTip().invalidate(reason: .actionPerformed)
         }
+
+        // Add observer for sign-in completion notification
+        let notificationName = NSNotification.Name(WordPressAuthenticationManager.WPSigninDidFinishNotification)
+        signinObserver = NotificationCenter.default.addObserver(forName: notificationName, object: nil, queue: nil) { [weak self] _ in
+            DispatchQueue.main.async {
+                self?.presentingViewController?.dismiss(animated: true)
+            }
+        }
+    }
+
+    deinit {
+        if let observer = signinObserver {
```

Comment:
(nit) it's redundant

```
f your app targets iOS 9.0 and later or macOS 10.11 and later, and you used [addObserver(_:selector:name:object:)](https://developer.apple.com/documentation/foundation/notificationcenter/1415360-addobserver), you do not need to unregister the observer. If you forget or are unable to remove the observer, the system cleans up the next time it would have posted to it.
```